lib/parse.js

```diff
@@ -14,6 +15,10 @@ exports = module.exports = function parse(content, options, isDocument) {
     content = content.toString();
   }
 
+  if (!isDocument) {
+    content = parseElement(content);
```

Comment:
This will always parse content using `parse5` and won't respect the `_useHtmlParser2` or `xmlMode` options (see right below).

One alternative implementation would be to always use `new Element` here, and initialize the parse5 extensions to empty objects:

```js
const elem = new Element(THE_TAG_NAME, {})
elem[ "x-attribsNamespace"] = {}
elem["x-attribsPrefix"] = {};
```

Reply:
True with xml it would generate element with incorrect namespace and htmlparser2 in general dont need it.

Now it will be co-parser only with parse5.

Problem with `new Element()` is parse5 expects every elements to have namespace and may fail to recognise them as html if it is missing.